### ISSUE TITLE
App frequently crashes when editing table data

### ISSUE THREAD

**Issue by clammish** [HeidiSQL/HeidiSQL#1904]
**Preconditions**
* HeidiSQL version: 12.6.0.6765
* Database type and version: MySQL Server 5.7.44, local server
* OS: Windows 11 (Arm). Parallels VM on macOS Sonoma 14.3.

**Describe the bug**
The app frequently crashes when editing table data manually using the data tab. The crash happens whenever the app would normally run an UPDATE statement, and after relaunching I must redo whatever I was just attempting to do.

As for whether the crash also happens when running UPDATE queries directly, I can't say since I don't really do that.

**To Reproduce**
I haven't been able to identify any pattern behind the crashes. Editing a field seems to result in a crash about 5% of the time. It does not seem to matter how long the app has been running.

**Screenshots or Crash reports**
Either no crash reports were generated or I was unable to find them.

**Comment by ansgarbecker** [comment]
You will need to provide a crash report here, which you can generate using the "save report" on the crash dialog.

And is it really the ARM version of Windows 11 you are running? I didn't knew normal 64bit applications run there.

**Comment by clammish** [comment]
There is no crash dialog. I hit enter to update something and the app just closes.

Regarding the architecture, the Settings app says "Parallels ARM Virtual Machine" and I have a "Program Files (Arm)" folder on my drive. I don't know whether Parallels or Microsoft is doing the heavy lifting, but everything seems to work quite well. (You are referring to the white popup with red, yellow and green buttons on it, correct? I encountered that thing previously when I was running into a separate issue, so it's not necessarily the OS that's preventing it from appearing here.)

**Comment by ansgarbecker** [comment]
I mean the madExcept dialog, here's a screenshot: http://madshi.net/madExceptDescription.htm

I'm still suspecting your virtual (ARM) environment to be causing random stuff here. However, I never got such a app termination here on a "normal" Windows installation.

I am not able to fix effects with neither a reproduction recipe nor a crash report.

**Comment by clammish** [comment]
Yep, that's the thing I saw earlier (when HeidiSQL was trying to load a corrupted SQL file) and which doesn't appear now.

I wouldn't be surprised if the app doesn't fully play nice with ARM Windows- when I couldn't get it to work through CrossOver I suspected I was in for a rough ride. I made this ticket mostly just to put this on your guys' radar. Unless you have an M-series Mac of your own I doubt you'll be able to repro, so feel free to recategorize this ticket as needed.
